docs/specs/appmodel.md

```diff
@@ -387,61 +386,60 @@ Interfaces allow Aspire to remain **open, flexible, and adaptable** as new types
 
 ## Resource Hierarchy and Parent-Child Relationships
 
-Aspire supports modeling **parent-child relationships** between resources to express ownership, containment, and grouping.
+Aspire supports modeling parent-child relationships between resources to express ownership, containment, and grouping.
 
-Parent-child relationships serve two purposes:
-- **Lifecycle Containment**: The child's execution is tied to the parent's — starting, stopping, and failures cascade from parent to child automatically.
-- **Dashboard Visualization**: The child appears **nested beneath** the parent in dashboards and visualizations, improving readability.
+Parent-child relationships primarily serve to:
+- Structural ownership and visualization: Expresses containment in the model and drives visual nesting in dashboards and visualizations.
+- Optional lifecycle coordination (opt-in): Some resource types choose to coordinate start/stop behavior relative to other resources. This behavior is separate from the parent-child structure and is not implied by any single interface.
 
----
-
-### Lifecycle Containment
+Aspire provides two ways to declare parent-child relationships:
 
-When a resource implements the `IResourceWithParent` interface, it declares **true containment** — meaning its lifecycle is controlled by its parent:
+**`IResourceWithParent` (static declaration)**: When a resource implements this interface, it statically declares a parent-child relationship. This establishes structural containment and visual nesting. `IResourceWithParent` implies `WithParentRelationship()`.
 
-- **Startup**: The child resource will only start after its parent starts (though readiness is independent).
-- **Shutdown**: If the parent is stopped or removed, the child is also stopped automatically.
-- **Failure Propagation**: If a parent enters a terminal failure state (`FailedToStart`, etc.), dependent children are stopped.
+**`WithParentRelationship()` (dynamic declaration)**: This method dynamically declares a parent-child relationship during resource construction, affecting only dashboard layout and visual grouping.
 
-> **Example:**  
-> A logging sidecar container is tied to the lifecycle of a main application container — if the main app stops, the logging sidecar is also terminated.
+Both approaches are purely structural and do not impose any lifecycle behavior.
```

Comment:
This makes sense to me as the desired implementation, but I'm not sure it reflects reality.  There are several reported bugs about resource lifecycles behaving differently when `WithParentRelationship` is used vs it not being used.

- https://github.com/dotnet/aspire/issues/9163
- https://github.com/dotnet/aspire/issues/9165
- https://github.com/dotnet/aspire/issues/10823